### PR TITLE
SWPBL-189734 -- As a user, I want to change block parameter values

### PR DESCRIPTION
**Pull request by KevinKimSonos**
## Overview

When an engineer is designing a system, they'll often make changes to one or block parameters. When a user clicks on a block, the list of parameters and their values for that block type should be editable in the multi-purpose panel. The value will be saved in its Block class that will be used to form a serialized format of the entire system.

## Done Criteria

- List of input text boxes appear next to each parameters values
- The input values are saved in its Block class structure
- Check data type and data constraints according to the parameters

## Notes

1. Able to change the parameter values and saved in the Block class.
2. Parameter value data types are shown as a hint in the according text fields.
3. Tested and verified that all other features work as expected.
4. Need to consider how to handle errors in the near future.
- When user types a value with wrong data type -> pop-up window showing that the value's data type is not allowed

***
See JIRA [here](https://jira.sonos.com/browse/SWPBL-189734).

### CODE REVIEW

File: example/Dynamite/context.h [hunk 1]
```diff
@@ -20,6 +20,7 @@ struct BlockNames
 struct BlockParameters
 {
     std::map<std::string, std::vector<std::string>> parameter_list;
+    std::map<std::string, std::vector<std::string>> parameter_type;
```
Comment (EvanShenkman-Sonos) [inline example/Dynamite/context.h:23]: Can we change these member names to something that indicates that they're maps? I don't have a great suggestion, but maybe something like `parameter_names_for_block`

**Review by EvanShenkman-Sonos — APPROVED** [review]
minor comment, otherwise it LGTM!

**Review by jehan-diaz — APPROVED** [review]

**Comment by EvanShenkman-Sonos** [comment]
> When I assign an invalid valid type to a parameter, the block disappears from view, as well as its inspector view. But I think the block is still part of the canvas because its port connections don't go away when the block disappears.

Hmm interesting, I don't see this behavior

**Comment by jehan-diaz** [comment]
> > When I assign an invalid valid type to a parameter, the block disappears from view, as well as its inspector view. But I think the block is still part of the canvas because its port connections don't go away when the block disappears.
> 
> Hmm interesting, I don't see this behavior

Interesting, I'm fine with merging anyway so I can start serializing, and I can take a look later again

**Comment by KevinKimSonos** [comment]
> > > When I assign an invalid valid type to a parameter, the block disappears from view, as well as its inspector view. But I think the block is still part of the canvas because its port connections don't go away when the block disappears.
> > 
> > 
> > Hmm interesting, I don't see this behavior
> 
> Interesting, I'm fine with merging anyway so I can start serializing, and I can take a look later again

Please let me know how to reproduce this behavior and I will look into it!

**Pull request merged**